op-batcher/batcher/channel_manager_test.go

```diff
@@ -483,3 +484,174 @@ func TestChannelManager_ChannelCreation(t *testing.T) {
+// set of market conditions. The test asserts that the DA type is changed at channel
+// submission time.
+func TestChannelManager_TxData(t *testing.T) {
+
+	type TestCase struct {
+		name                            string
+		chooseBlobsWhenChannelCreated   bool
+		chooseBlobsWhenChannelSubmitted bool
+	}
+
+	tt := []TestCase{
+		{"blobs->blobs", true, true},
+		{"calldata->calldata", false, false},
+		{"blobs->calldata", true, false},
+		{"calldata->blobs", false, true},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			l := testlog.Logger(t, log.LevelCrit)
+
+			cfg := newFakeDynamicEthChannelConfig(l, 1000)
+
+			cfg.chooseBlobs = tc.chooseBlobsWhenChannelCreated
+			m := NewChannelManager(l, metrics.NoopMetrics, cfg, defaultTestRollupConfig)
+			require.Equal(t, tc.chooseBlobsWhenChannelCreated, m.defaultCfg.UseBlobs)
+
+			// Seed channel manager with a block
+			rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+			blockA := derivetest.RandomL2BlockWithChainId(rng, 10, defaultTestRollupConfig.L2ChainID)
+			m.blocks = []*types.Block{blockA}
+
+			// Call TxData a first time to trigger blocks->channels pipeline
+			_, err := m.TxData(eth.BlockID{})
+			require.ErrorIs(t, err, ErrInsufficientData)
+
+			// The test requires us to have something in the channel queue
+			// at this point, but not yet ready to send and not full
+			require.NotEmpty(t, m.channelQueue)
+			require.False(t, m.channelQueue[0].IsFull())
+
+			// Simulate updated market conditions
+			// by possibly flipping the state of the
+			// fake channel provider
+			l.Info("updating market conditions", "chooseBlobs", tc.chooseBlobsWhenChannelSubmitted)
+			cfg.chooseBlobs = tc.chooseBlobsWhenChannelSubmitted
+
+			// Add a block and call TxData until
+			// we get some data to submit
+			var data txData
+			for {
+				m.blocks = []*types.Block{blockA}
+				data, err = m.TxData(eth.BlockID{})
+				if err == nil && data.Len() > 0 {
+					break
+				}
+				if !errors.Is(err, ErrInsufficientData) {
+					require.NoError(t, err)
+				}
+			}
+
+			require.Equal(t, tc.chooseBlobsWhenChannelSubmitted, data.asBlob)
+			require.Equal(t, tc.chooseBlobsWhenChannelSubmitted, m.defaultCfg.UseBlobs)
+		})
+	}
+
+}
+
+// TestChannelManager_Requeue sees the channel manager with blocks,
+// takes a state snapshot, triggers the blocks->channels pipeline,
+// and then calls Requeue. Finally, it asserts the channel manager's
+// state is equal to the snapshot. It repeats this for a channel
+// which has a pending transaction and verifies that Requeue is then
+// a noop.
+func TestChannelManager_Requeue(t *testing.T) {
+	l := testlog.Logger(t, log.LevelCrit)
+	cfg := channelManagerTestConfig(100, derive.SingularBatchType)
+	m := NewChannelManager(l, metrics.NoopMetrics, cfg, defaultTestRollupConfig)
+
+	// Seed channel manager with blocks
+	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
Probably better to fix this to make the test more deterministic.

Reply:
agree, not sure if we benefit from a random seed

Reply:
fixed in [2a9b7ef](https://github.com/ethereum-optimism/optimism/pull/12002/commits/2a9b7efd1d58d3d70a255c889e8b3552df59239f)